### PR TITLE
Replace common directory in multiplatform structure

### LINKED ISSUE #123: Update Documentation for INIT
The documentation needs to clearly outline the differences between a Landscape, Immersive and Components application.  Further we need to make clear what choosing Lumin, iOS and Android means for Components applications.

### LINKED ISSUE #141: Place common directory from multiplatform structure in reactnative directory
Check if it's possible, if so, place common directory from multiplatform structure in reactnative directory and resolve imports properly

### LINKED ISSUE #144: Add proper .gitignore configuration for multiplatform components structure

### LINKED ISSUE #145: Fix problem with symlink dependencies for Lumin in multiplatform components structure
If lumin has symlink as a dependency, the rollup plugin creates wrong files structure

### PR DESCRIPTION
**Pull request by panwrona**
- Replace common directory in multiplatform structure: Resolve #141 
- Add proper gitignore file: Resolve #144 
- Fix lumin local dependency symlink issue: Resolve #145 
- Update documentation for init: Resolve #123

### CODE REVIEW

File: template_overlay_typescript_components/reactnative/package.json [hunk 1]
```diff
@@ -16,14 +16,12 @@
...
-    "typescript": "^3.6.3",
-    "magic-script-typings": "1.4.0"
+    "typescript": "^3.6.3"
   },
   "dependencies": {
-    "magic-script-polyfills": "2.4.2",
     "react": "16.8.3",
     "react-native": "0.60.5",
-    "magic-script-components-react-native": "1.0.1"
+    "magic-script-components-react-native": "1.0.3"
```
Comment (kpiascik) [inline template_overlay_typescript_components/reactnative/package.json]: this should be `^1.0.3`

File: template_multiplatform_components/reactnative/package.json [hunk 2]
```diff
@@ -10,11 +10,9 @@
...
     "metro-react-native-babel-preset": "^0.54.1",
-    "react-test-renderer": "16.8.3",
-    "magic-script-typings": "1.4.0"
+    "react-test-renderer": "16.8.3"
   },
   "dependencies": {
-    "magic-script-polyfills": "2.4.2",
     "react": "16.8.3",
     "react-native": "0.60.5",
     "magic-script-components-react-native": "1.0.1"
```
Comment (kpiascik) [inline template_multiplatform_components/reactnative/package.json]: please update the version here as well.

File: template_overlay_typescript_components/reactnative/package.json (hunk 1 above)
Comment (grozdanov) [reply template_overlay_typescript_components/reactnative/package.json]: Updated !

File: template_multiplatform_components/reactnative/package.json (hunk 2 above)
Comment (grozdanov) [reply template_multiplatform_components/reactnative/package.json]: Version has been updated to ^1.0.3

**Pull request merged**